### PR TITLE
Tag algorithm buffers with parameter they are tracking.

### LINKED ISSUE #986: Statistics on algorithm buffers are hard to monitor
They generally only get a role but no other identifying information.

Often they mirror a specific parameter attached to a specific brick, so I'd recommend:

* Annotating them with the relevant brick
* Adding a `for_parameter` key to the tag similar to `replacement_of`.
* Maybe letting the `Selector` search for them somehow.

### PR DESCRIPTION
**Pull request by dwf**
Fixes #986.

### CODE REVIEW

File: blocks/algorithms/__init__.py [hunk 1]
```diff
@@ -801,10 +801,10 @@ def __init__(self, learning_rate=0.002,
             add_role(param, ALGORITHM_HYPERPARAMETER)
 
     def compute_step(self, parameter, previous_step):
-        mean = shared_floatx_zeros_matching(parameter, 'mean')
-        add_role(mean, ALGORITHM_BUFFER)
-        variance = shared_floatx_zeros_matching(parameter, 'variance')
-        add_role(variance, ALGORITHM_BUFFER)
+        mean = _create_algorithm_buffer_for(parameter, 'mean')
+        mean.tag.for_parameter = parameter
```
Comment (rizar) [inline blocks/algorithms/__init__.py]: This line seems to be not necessary, as well as a similar one below.

Comment (dwf) [inline blocks/algorithms/__init__.py]: Good point, these predate my refactoring. I will fix them up later this evening (off to dinner).

**Comment by dwf** [comment]
@rizar Fixed the issue you pointed out.

However, it also looks like Travis is shitting its pants (no g++?).

**Comment by rizar** [comment]
LGTM on my side, can't help you with Travis though.

**Comment by dwf** [comment]
So that coverage number is clearly bullshit caused by the Travis incident, since you gave it the LGTM I'll merge.

**Pull request merged**